File: Data Structures & Algorithms/three-integer-sum/submission-1.py
class Solution:
    def threeSum(self, nums: List[int]) -> List[List[int]]:
        n = len(nums)
        ans = []

        for i in range(n-2):
            for j in range(i+1,n-1):
                cur = nums[i]+nums[j]
                if -cur in nums[j+1:]:
                    triplet = sorted([nums[i],nums[j],-cur])
                    if triplet not in ans:
                        ans.append(triplet)
        
        return ans

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.cpp
#include <vector>
using namespace std;

class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        int product = 1, zeros = 0;
        for(int i : nums){
            if(i != 0){
                product *= i;
            }else{
                zeros++;
            }
        }

        if(zeros>1){return vector<int>(nums.size(), 0);}

        vector<int> ans;
        if(zeros==1){
            for(int i : nums){
                ans.push_back(i != 0 ? 0 : product);
            }
        }else{
            for(int i : nums){
                ans.push_back(product/i);
            }
        }

        return ans;
    }
};
